hiplot/fetchers.py

```diff
@@ -47,6 +77,16 @@ def __call__(self, uri: str) -> hip.Experiment:
             return hip.Experiment.merge({v: load_xp_with_fetchers(self.fetchers, v) for v in defs})
         return hip.Experiment.merge({k: load_xp_with_fetchers(self.fetchers, v) for k, v in defs.items()})
 
+    def partial(self, uri: str) -> int:
+        """
+        Returns the end position of the multi block
+        """
+        if not uri.startswith(self.MULTI_PREFIX):
+            raise hip.ExperimentFetcherDoesntApply()
+        decoder = json.JSONDecoder()
+        _, current_read_offset = decoder.raw_decode(uri[len(self.MULTI_PREFIX):])
+        return len(self.MULTI_PREFIX) + current_read_offset
+
```

Comment:
I'm not very happy with the name of this function. Do you have a better idea for that?

Reply:
fair enough, partial is not a very good name since it has a very different meaning in the standard library :D
Why not `get_json_offset`?

Reply:
This should be generic to any experiment fetcher. It should return the last offset parsed by the experiment fetcher.
In the case of the multi block, it will return the end of the JSON object, but it's only a special case.

Reply:
It would feel more self contained if you returned onset and offset together, as something like `get_string_span` or `get_string_position`

Reply:
or if you have the onset somewhere else, `get_string_length` ?

Reply:
Onset is always zero. I'll go for `get_uri_length,  sounds better thanks :)